pymc3/distributions/multivariate.py

```diff
@@ -1091,3 +1091,183 @@ def logp(self, x):
                      eta > 0,
                      broadcast_conditions=False
         )
+
+
+class MatrixNormal(Continuous):
+    R"""
+    Matrix-valued normal log-likelihood.
+
+    ===============  =====================================
+    Support          :math:`x \in \mathbb{R}^{q \times p}`
+    Mean             :math:`\mu`
+    Right Variance   :math:`T_R^{-1}`
+    Left Variance    :math:`T_L^{-1}`
+    ===============  =====================================
+
+    Parameters
+    ----------
+    mu : array
+        Vector of means.
+    rcov : array
+        Right (or row) pxp covariance matrix. Defines variance within rows.
+        Exactly one of rcov or rchol is needed.
+    rchol : array
+        Cholesky decomposition of pxp covariance matrix. Defines variance
+        within rows. Exactly one of rcov or rchol is needed.
+    lcov : array
+        Left (or column) qxq covariance matrix. Defines variance within
+        columns. Exactly one of lcov or lchol is needed.
+    lchol : array
+        Cholesky decomposition of qxq covariance matrix. Defines variance
+        within columns. Exactly one of lcov or lchol is needed.
+
+    Examples
+    --------
+    ?
+    """
+
+    def __init__(self, mu=0, rcov=None, rchol=None, rtau=None,
+                 lcov=None, lchol=None, ltau=None, *args, **kwargs):
+
+        self.setup_matrices(rcov, rchol, rtau, lcov, lchol, ltau)
+
+        shape = kwargs.pop('shape', None)
```

Comment:
I don't see a problem in making the requiring the shape argument. Just raise a `TypeError` if you don't get one. (I guess in py3 we could make that a required keyword-only argument...)